src/ui_contextmenu.cpp

```diff
@@ -293,7 +300,79 @@ class OpenLyricsContextSubItem : public contextmenu_item_simple
                                                  core_api::get_main_window(),
                                                  "Marking tracks as instrumental...");
             } break;
+            case cmd_seek_to_next_lyric: {
+                auto& panels = get_active_panels();
+                if (!panels.empty())
+                {
+                    LyricPanel* panel = panels[0];
+                    LyricData lyrics = panel->get_lyrics();
+                    service_ptr_t<playback_control> playback = playback_control::get();
+
+                    double timeNow = playback->playback_get_position();
+
+                    for (size_t i = 0; i < lyrics.lines.size(); ++i)
+                    {
+                        double ts = lyrics.lines[i].timestamp;
+                        if (ts > timeNow)
+                        {
+                            playback->playback_seek(ts);
+                            break;
+                        }
+                    }
+                }
+                break;
+            }
+            case cmd_seek_to_prev_lyric: {
+                auto& panels = get_active_panels();
+                if (!panels.empty())
+                {
+                    LyricPanel* panel = panels[0];
+                    LyricData lyrics = panel->get_lyrics();
+                    service_ptr_t<playback_control> playback = playback_control::get();
+
+                    bool skipOne = true;
+                    double timeNow = playback->playback_get_position();
+
+                    for (size_t i = lyrics.lines.size(); i-- > 0; )
+                    {
+                        double ts = lyrics.lines[i].timestamp;
+                        if (ts < timeNow)
+                        {
+                            if (skipOne) 
+                            {
+                                skipOne = false;
+                                continue;
+                            }
+                            playback->playback_seek(ts);
+                            break;
+                        }
+                    }
+                }
+                break;
+            }
+            case cmd_seek_to_repeat_current_lyric: {
+                auto& panels = get_active_panels();
+                if (!panels.empty())
+                {
+                    LyricPanel* panel = panels[0];
+                    LyricData lyrics = panel->get_lyrics();
+                    service_ptr_t<playback_control> playback = playback_control::get();
 
+                    double timeNow = playback->playback_get_position();
+
+                    for (size_t i = lyrics.lines.size(); i-- > 0; )

```

Comment:
Looks like this is just `std::lower_bound`, could we use that to simplify all of these lookups?